### DISCUSSION TITLE
Lazy hydration

### DISCUSSION THREAD

**Discussion by thanhtoan1196** [vercel/next.js#28445] [Ideas]
### Describe the feature you'd like to request

Using lazy hydration to delay the hydration of pre-rendered HTML

### Describe the solution you'd like

Delay the hydration of server-side rendered HTML until it’s really needed

### Describe alternatives you've considered

Lazy Hydration of Server-Side Rendered Vue.js Components
https://www.npmjs.com/package/vue-lazy-hydration

**Comment by timneutkens** [comment]
We're working with the React team to land selective hydration: https://github.com/reactwg/react-18/discussions/37

**Reply by solarkraft** [reply]
What has come of this? Is it now possible to selectively hydrate statically pre-rendered components, loading their hydration code when needed?

**Reply by Shramkoweb** [reply]
@timneutkens any updates?

**Reply by timneutkens** [reply]
You'll want to follow https://nextjs.org/blog/layouts-rfc which leverages React Server Components to have Selective Hydration by default.

**Reply by Shramkoweb** [reply]
> selectively hydrate


I don't think it's relevant, React Server Component != Selectively hydrate.
Can u help find information on the work of Selective hydrate with static render html ?
Because i think its not work https://github.com/reactwg/react-18/discussions/37 in Next.js + SSG ?

**Reply by timneutkens** [reply]
> I don't think it's relevant, React Server Component != Selectively hydrate.

It's very relevant to Server Components as it changes the way bundling works. All of these individual features tie back into Server Components, they're building blocks that allow for Server Components to work.

Server Components does not mean running a server btw, it supports static generation too.

Wrapping a suspense boundary at the root of your app will opt-in the hydration to concurrent rendering but it doesn't have the optimized bundling strategy so you'd still end up waiting on all components to load before hydration starts.

My recommendation is still to follow https://nextjs.org/blog/layouts-rfc as it's the way we're adopting all React 18 features including concurrent rendering, selective hydration, etc.

**Reply by jaskome** [reply]
@timneutkens after your comment and some tests with dynamic imports, I realized that selective hydration does not work as expected with nextjs. But this blog post: https://vercel.com/blog/improving-interaction-to-next-paint-with-react-18-and-suspense  suggests that selective hydration should work. I also tested it on nextjs.org, and I think also there the hydration starts after all scripts are loaded?! Do I misunderstand something in this blog post or your comment?

**Reply by loliwe1** [reply]
Any updates? I’m having the same issue — hydration only starts after all JS chunks are downloaded
